middleware/src/handlers/handlers.go

```diff
@@ -0,0 +1,87 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	//"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+	//"github.com/tidwall/gjson"
+	"github.com/digitalbitbox/bitbox-base/middleware/src"
+	"github.com/digitalbitbox/bitbox-base/middleware/util/logging"
+	"github.com/sirupsen/logrus"
+)
+
+type Middleware interface {
+	Start() <-chan *middleware.MiddlewareInfoStruct
+}
+
+// Handlers provides a web api
+type Handlers struct {
+	Router     *mux.Router
+	clients    map[*websocket.Conn]bool
+	upgrader   websocket.Upgrader
+	log        *logrus.Entry
+	middleware Middleware
+	//TODO: In future this event should have a generic interface (thus only containing raw json)
+	middlewareEvents <-chan *middleware.MiddlewareInfoStruct
+}
+
+func NewHandlers(middlewareInstance Middleware) *Handlers {
+	router := mux.NewRouter()
+
+	handlers := &Handlers{
+		middleware: middlewareInstance,
+		Router:     router,
+		clients:    make(map[*websocket.Conn]bool),
+		upgrader: websocket.Upgrader{
+			CheckOrigin: func(r *http.Request) bool {
+				return true
+			},
+		},
+		log: logging.Get().WithGroup("Handlers"),
+	}
+	handlers.Router.HandleFunc("/", handlers.rootHandler).Methods("GET")
+	handlers.Router.HandleFunc("/ws", handlers.wsHandler)
+
+	handlers.middlewareEvents = handlers.middleware.Start()
+	go handlers.echo()
+	return handlers
+}
+
+func (handlers *Handlers) rootHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Maybe we could add a comment as to what `rootHandler` does? (And whether it's expected to remain, or if it's a temporary / testing thing.)